kind/common.sh

```diff
@@ -65,11 +65,17 @@ aarch64* | arm64*)
     ;;
 esac
 
+# check os
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    SED="gsed"
+else
+    SED="sed"
```

Comment:
this if else branch LGTM, I will leave to @marceloamaral to have a review for his comments.

Reply:
What is the problem with sed?

My macOS does not have gsed by default.. Can't we just use sed?

Reply:
@vprashar2929 , @marceloamaral , I suppose it's related mac and linux has different behavior for sed.
https://stackoverflow.com/questions/4247068/sed-command-with-i-option-failing-on-mac-but-works-on-linux
https://stackoverflow.com/questions/33906892/using-sed-i-mac-osx-without-a-backup-file
https://superuser.com/questions/307165/newlines-in-sed-on-mac-os-x
I try to find some solutions... FYI

Reply:
Absolutely sed behaves differntly on Mac os. I tried using `sed -i.bak -e` as well but same result. It was creating files with `.bak` at the end😅. Only `gsed` was working in this case

Reply:
Let's try to fix and use `sed` only. Otherwise we will create a dependency to install a new tool.

Reply:
@marceloamaral I initially tried the same approach. But the sed command made `prometheus-prometheus.yaml''` file. And kubectl again failed to create the manifest
Error snippet:
```
+ kubectl create -f 'kube-prometheus/manifests/prometheus-prometheus.yaml'\'''\'''
Error from server (AlreadyExists): error when creating "kube-prometheus/manifests/prometheus-prometheus.yaml''": prometheuses.monitoring.coreos.com "k8s" already exists
```
`ls` on `kubeprometheus/manifests`
```
ls -ltr kube-prometheus/manifests/prometheus-prometheus.*
-rw-r--r--  1 vibhuprashar  staff  1238 May 22 13:38 kube-prometheus/manifests/prometheus-prometheus.yaml''
-rw-r--r--  1 vibhuprashar  staff  1238 May 22 13:38 kube-prometheus/manifests/prometheus-prometheus.yaml
```

I agree that if we use `gsed` then it will create an additional dependency to be installed. Apart from `gsed` I can only see another way to make the script remove the manifests filenames with `-e` or `''` before kubectl creates them. wdyt?

Reply:
@marceloamaral [This](https://github.com/sustainable-computing-io/local-dev-cluster/pull/9/commits/37091d3547b57fffdac8ff30d6017b5e4e1673b9) might be a better approach as compared to making script delete backup files. wdyt?